jazz_assets/components/validation.js

```diff
@@ -0,0 +1,313 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+/**
+    Helper Validation functions for Deployments-Catalog
+    @module: validate.js
+    @description: Defines validate functions.
+    @author:
+    @version: 1.0
+**/
+
+const logger = require("./logger.js")(); //Import the logging module.
+const _ = require("lodash");
+const crud = require("./crud")();
+const utils = require("./utils.js");
+
+function validateIsEmptyInputData(assets_data) {
+    logger.debug("Inside validateIsEmptyInputData");
+    return new Promise((resolve, reject) => {
+        if (Object.keys(assets_data).length === 0) {
+            reject({
+                "result": "inputError",
+                "message": "Input asset data cannot be empty"
+            });
+        } else {
+            resolve({
+                "result": "success",
+                "input": assets_data
+            });
+        }
+    });
+};
+
+function validateEmptyFieldsVal(assets_data) {
+    logger.debug("Inside validateEmptyFieldsVal");
+    return new Promise((resolve, reject) => {
+        var invalid_fields = [];
+        Object.keys(assets_data).forEach((field) => {
```

Comment:
well replacing map with foreach is not what i meant and that is not always what you should do. i think i had suggested using filter may be. Just get the values from assets_Data where there is some value. What do you think?  
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
Using filter